spec/validations/svg_spec.rb

```diff
@@ -96,99 +92,116 @@
 
-    malformed_markup_examples.map do |markup|
-      let(:markup) { markup }
-      let(:empty_svg) { %(<svg/>) }
+        errors = ZendeskAppsSupport::Validations::Svg.call(package)
 
-      it 'have the contents of their svg tags emptied, with a warning that the offending svgs were modified' do
-        errors = subject.call(package)
-        expect(IO).to have_received(:write).with(svg.absolute_path, empty_svg)
-        expect(package.warnings[0]).to eq(warning)
+        empty_svg = %(<svg/>)
+        expect(IO).to have_received(:write).with('/tmp/icon.svg', empty_svg)
+        expect(package.warnings[0]).to eq(
+          'The markup in assets/icon.svg has been edited for use in Zendesk and may not display as intended.'
+        )
         expect(errors).to be_empty
       end
     end
   end
 
   context 'svgs that contain embedded bitmap(s)' do
-    let(:markup) do
-      %(<svg viewBox="0 0 26 26" id="1px-white-square" width="100%" height="100%"><image width="1" height="1" x="0" \
-y="0" xlink:href="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs="/></svg>)
-    end
-    let(:warning) do
-      %(assets/icon_nav_bar.svg contains an embedded bitmap and cannot be used as an app icon. It has been replaced \
-with a default placeholder icon.)
-    end
-
-    before do
-      allow(subject).to receive(:rewrite_svg).and_call_original
-    end
-
     it 'generate a warns the developer, and overwrites the svg with a placeholder image' do
-      errors = subject.call(package)
-      expect(subject).to have_received(:rewrite_svg).with(svg, subject::PLACEHOLDER_SVG_MARKUP, package, errors)
-      expect(package.warnings[0]).to eq(warning)
+      markup = %(<svg viewBox="0 0 26 26" id="1px-white-square" width="100%" height="100%"><image width="1" height="1" x="0" \
+y="0" xlink:href="data:image/gif;base64,R0lGODlhAQABAIAAAP///wAAACwAAAAAAQABAAACAkQBADs="/></svg>)
+      svg_file = double('SVGFile', read: markup, relative_path: 'assets/icon.svg', absolute_path: '/tmp/icon.svg')
+      package = mock_package(svg_file)
+
+      errors = ZendeskAppsSupport::Validations::Svg.call(package)
+
+      placeholder_markup = %(<?xml version="1.0" encoding="utf-8"?><svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" \
+xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px" viewBox="0 0 18 18" style="enable-background:new 0 0 18 18;" xml:space="preserve">\
+<path id="Fill-3" d="M1.5,6.1C1.3,5.9,1,6,1,6.4l0,7c0,0.3,0.2,0.7,0.5,0.9l6,3.6C7.8,18.1,8,18,8,17.6l0-7.1c0-0.3-0.2-0.7-0.5-0.9 L1.5,6.1z"/>\
+<path id="Fill-5" d="M10.5,17.9c-0.3,0.2-0.5,0-0.5-0.3l0-7c0-0.3,0.2-0.7,0.5-0.9l6-3.6C16.8,5.9,17,6,17,6.4l0,7.1 c0,0.3-0.2,0.7-0.5,0.9L10.5,17.9z"/>\
+<path id="Fill-1" d="M2.2,3.7c-0.3,0.2-0.3,0.4,0,0.6l6.2,3.6C8.7,8,9.2,8,9.4,7.9l6.3-3.6c0.3-0.2,0.3-0.4,0-0.6L9.5,0.1 C9.2,0,8.8,0,8.5,0.1L2.2,3.7z"/></svg>)
+      expect(IO).to have_received(:write).with('/tmp/icon.svg', placeholder_markup)
```

Comment:
Why is it receiving `write` now instead of `rewrite_svg`?

Reply:
I rewrote the test to not spy on private methods. As a best practice, we should test behaviour over implementation.